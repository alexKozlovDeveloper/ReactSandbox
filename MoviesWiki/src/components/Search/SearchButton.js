import React, { Component } from "react";

import styles from "../../styles/SearchButton.css"

const SearchButton = ({ buttonText, updateFunc }) => {
    return  <button className={styles.button} onClick={() => updateFunc()}>{buttonText}</button>;
}

export default SearchButton;