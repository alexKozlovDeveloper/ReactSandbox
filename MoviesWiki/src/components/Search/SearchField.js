import React, { Component } from "react";

import styles from "../../styles/SearchField.css";

const SearchField = ({ placeHolder, updateFunc, submitFunc }) => {
    return <input className={styles.searchfield} type="text" placeholder={placeHolder} onInput={(e) => updateFunc(e.currentTarget.value)} onKeyDown={(e) => submitFunc(e)} />;
}

export default SearchField;