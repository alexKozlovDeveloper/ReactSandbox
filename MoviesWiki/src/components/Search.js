import React, { Component } from "react";

import SearchField from "./SearchField";
import SearchButton from "./SearchButton";
import SearchFilter from "./SearchFilter";

import styles from "../styles/Search.css"

class Search extends Component {
    render() {
        return (            
            <div className={styles.search}>                
                <div>                
                    <SearchField />
                    <SearchButton />
                </div>
                <SearchFilter />
            </div>           
        );
    }
}

export default Search;