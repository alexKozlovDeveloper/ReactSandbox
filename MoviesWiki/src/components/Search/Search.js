import React, { Component } from "react";

import SearchField from "./SearchField";
import SearchButton from "./SearchButton";
import CustomOptionList from "../Common/CustomOptionList";

import styles from "../../styles/Search.css"

class Search extends Component {
    render() {
        return (            
            <div className={styles.search}> 
                <div>                
                    <SearchField placeHolder={this.props.config.placeHolder}/>
                    <SearchButton buttonText={this.props.config.buttonText}/>
                </div>
                <CustomOptionList config={this.props.config.searchFilter}/>
            </div>           
        );
    }
}

export default Search;