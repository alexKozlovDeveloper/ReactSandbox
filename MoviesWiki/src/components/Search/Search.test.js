import React from 'react';
import { shallow, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import Search from "./Search";

configure({adapter: new Adapter()});

test('Search rendering test', () => {

    var searchConfig = {
        placeHolder: "Quentin Tarantino",
        buttonText: "Search",
        searchFilter: {
            title: "Search by",
            selectedIndex: "0",
            options: ["Title", "Genre"]
        }
    }

    const component = shallow(<Search config={searchConfig}/>);
    expect(shallowToJson(component)).toMatchSnapshot();
 });

