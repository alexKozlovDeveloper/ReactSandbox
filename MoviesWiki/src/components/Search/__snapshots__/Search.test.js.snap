// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search rendering test 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Connect(Search)
    config={
      Object {
        "buttonText": "Search",
        "placeHolder": "Quentin Tarantino",
        "searchFilter": Object {
          "options": Array [
            "Title",
            "Genre",
          ],
          "selectedIndex": "0",
          "title": "Search by",
        },
      }
    }
  />
</ContextProvider>
`;
